chemistry/structure.py

```diff
@@ -919,11 +919,19 @@ def update_dihedral_exclusions(self):
         turns it to `True` if the two end atoms are in the bond or angle
         partners arrays
         """
+        list14 = []
```

Comment:
Use `set` here instead of `list`.  Object lookup in a set is O(1) average case.  In a list it's O(N).